style(tokens): new token sets for core, semantic, and branding

ju-Skinner opened the pull request:
# Description

Adds semantic tokens to our Design System, leveraging both Token Studio and Style Dictionary

Fixes #(issue)

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [ ] unit tests
- [ ] e2e tests
- [ ] accessibility tests
- [x] tested manually
- [x] other:


# Checklist:

If not applicable, leave options unchecked.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

dev-kjbot committed:
style: create new core tokenset

dev-kjbot committed:
style: change font-weight to use numerical values in core tokenset

dev-kjbot committed:
style: adds half of semantic tokens

dev-kjbot committed:
style: adds the second half of the semantic tokens

dev-kjbot committed:
style: add kajabi products themes and mode (light and dark)

ju-Skinner committed:
feat: add sd-transformations to generate tokens from token-studio

dev-kjbot committed:
style: update themes to include component/avatar

dev-kjbot committed:
style: remove old core and semantic tokens

they have been moved to the tokens/base folder

ju-Skinner committed:
style(transform): refactor transform scripts to be more modular

dev-kjbot committed:
style: changes for brand semantic and modes

dev-kjbot committed:
style(tokens): working version of token studio to figma variables with token names in values

dev-kjbot committed:
style(tokens): add color.brand to kajabi semantic tokens

dev-kjbot committed:
style(tokens): cleanup semantic tokens and a few core tokens for dimensions

ju-Skinner committed:
style(tokens): newly generated tokens from style dictionary

this is temporary because we'd like to have this as a separate library

ju-Skinner committed:
style(tokens): restructure semantic brand token names and locations

ju-Skinner committed:
style(button): update styles to use semantic tokens

dev-kjbot committed:
style(token-studio): update tokens

ju-Skinner committed:
style(tokens): generate new style sheets from sd-transforms

ju-Skinner committed:
style(avatar): clean up testing of component semantic level tokens

ju-Skinner committed:
style(application): add a the brand semantic style sheet for testing

dev-kjbot committed:
style(token-studio): remove border-radius default and none

dev-kjbot committed:
style(token-studio): add font-family semantic tokens

dev-kjbot committed:
style(token-studio): adds color.background.container.hover semantic token

ju-Skinner committed:
style(tokens): update the semantic tokens by running build.tokens

dev-kjbot committed:
style(token-studio): add line-spacing for headings and new dimension token for 2px

ju-Skinner committed:
style(button): change styles to use semantic tokens (#288)

ju-Skinner committed:
style(checkbox): change or remove CSS Custom props to use semantic tokens (#289)

ju-Skinner committed:
style(divider): change tokens to use new semantic tokens from token studio (#291)

Commit:
style(copytext): update copytext tokens (#293)

* style: updates copy text with semantic tokens

* chore: fixes linting issues

dev-kjbot committed:
style(token-studio): add fallback for font-families Greet and Inter

Commit:
style(accordion): update accordion tokens (#292)

* style: updates accordion with semantic tokens

* chore: fixes linting issues

ju-Skinner committed:
style(token-studio): update sentiment colors to *-100 colors

we will need to update pds-button to override to 500 token values

Commit:
style: updates component with semantic tokens (#297)

Commit:
style: updates component with semantic tokens (#298)

ju-Skinner committed:
style(input): update input component to use new semantic tokens (#299)

ju-Skinner committed:
style(loader): change css properties to use semantic tokens (#300)

ju-Skinner committed:
style(progress): change progress css custom props to become semantic tokens (#301)

Commit:
style(tooltip): updates component with semantic tokens (#302)

ju-Skinner committed:
style(radio): change radio component to use new semantic tokens (#303)

Commit:
style(tabs): updates component with semantic tokens (#306)

* style(tabs): updates component with semantci tokens

* style: updates filter background token

Commit:
style(switch): updates component with semantic tokens (#308)

ju-Skinner committed:
style(box): use semantic tokens in box stylesheet (#304)

ju-Skinner committed:
style(sortable): change tokens to use semantic tokens (#305)

ju-Skinner committed:
style(table): change css custom props to use semantic tokens (#309)

ju-Skinner committed:
style(chip): change css props to use semantic tokens (#312)

Commit:
chore: removes base css file from components (#313)

ju-Skinner committed:
style(avatar): clean up missing semantic tokens

ju-Skinner committed:
style(button): change text color to match their variant

ju-Skinner committed:
style(token-studio): change sentiment colors to use *-500

dev-kjbot committed:
style(token-studio): create semantic tokens for chip

ju-Skinner committed:
style(token-studio): refactor component generator

dev-kjbot committed:
style(tokens-studio): remove avatar semantic tokens

ju-Skinner committed:
style(tokens): refactor sd-tranforms to read component names from token/components

ju-Skinner committed:
style(box): refactor to use new semantic tokens (#319)

pixelflips committed:
fix: update incorrect core color token values (#322)

* fix: adjust color token values

* fix: adjust missed blue values

* style(tokens): new colors

---------

Co-authored-by: Julian Skinner <ju.skinner@gmail.com>

Commit:
style: updates some components with semantic tokens (#325)

* style: updates some components with semantic tokens

* style(chip): update chip to use component semantic tokens

* style(input): create new component specific semantic tokens

created input.json for tokenstudio

updated both the themes and metadata.json files

---------

Co-authored-by: Julian Skinner <ju.skinner@gmail.com>

ju-Skinner committed:
style(tokens-studio): update background container disabled token

ju-Skinner committed:
style(input): sync styles for readonly and disabled label and input elements

ju-Skinner committed:
style(textarea): sync styles for readonly and disabled label and textarea

ju-Skinner committed:
refactor(label): moves label to a style utils

ju-Skinner committed:
style(textarea): refactor to use utils/label.scss

ju-Skinner committed:
style(fonts): add Greet Condensed font family

pixelflips commented on libs/core/src/components/pds-accordion/pds-accordion.scss:
```diff
@@ -3,29 +3,14 @@
-  --typography-default: var(--pine-typography-body-md-medium);
 
-  border-radius: var(--border-radius-default);
+  border-radius: var(--pine-dimension-xs);
```
Should this be a border-radius token?

ju-Skinner replied:
See comment below

pixelflips commented on libs/core/src/components/pds-checkbox/pds-checkbox.scss:
```diff
@@ -82,30 +62,30 @@
-  height: var(--sizing-input);
+  background-color: var(--pine-color-secondary);
+  border: var(--pine-border);
+  border-radius: var(--pine-dimension-2xs);
```
I am confused by the `dimension` token, is this replacing spacing? Also in some areas, a radius token (e.g. `--pine-border-radius-round`) is used and in others its dimensions (e.g. `--pine-dimension-2xs`).

ju-Skinner replied:
@phillip yes, the dimension token now covers a few areas. See [here](https://docs.tokens.studio/token-types/token-type-dimension) for more details.

As for ...radius-round and `radius-circle` those are special cases so we called those out.

pixelflips commented on libs/core/src/components/pds-chip/pds-chip.scss:
```diff
@@ -196,31 +161,31 @@ $pds-chip-sentiment-hover: (
-    padding-block: var(--spacing-xs);
-    padding-inline: var(--spacing-xxs);
+    padding-block: var(--pine-dimension-2xs);
+    padding-inline: var(--pine-dimension-025);
```
Are dimensions tshirt sizing or a numeric scale?

anechol replied:
T-shirt sizing is only for a small defined set of semantic tokens. Core tokens are still numeric. This particular token has no corresponding semantic token, so we defaulted back to core.

pixelflips replied:
Not a blocker, but maybe we think about having these assigned to a component token, just so it's obvious it's a one-off.

pixelflips commented on libs/core/src/global/styles/tokens/base/_core.scss:
```diff
@@ -1,3 +1,7 @@
+ */
+
 :root {
   --pine-border-radius-0: 0px;
```
Is it possible to remove the unit (px) from the 0 values. 0 is usually unitless.

ju-Skinner replied:
This is a style dictionary thing. We don't have control over that.

pixelflips reviewed (commented):
I have a few questions, but this is off to an awesome start. If anything is unclear or I am misunderstanding, jut let me know.

ju-Skinner committed:
test: fix failling specs for readonly components

pixelflips reviewed (approved):
Thanks for the clarifications. LGTM!

QuintonJason reviewed (approved):
Good to go. Great work!

anechol reviewed (approved):
Since we tackled this together, my approval is biased, but thanks for keeping this PR in line through all our changes.

ju-Skinner committed:
style: update line-heights to percentage

StyleDictionary handles the conversion to unitless values

whereas figma requires percentages

ju-Skinner committed:
style: update Font Family's to match the system installed font name

dev-kjbot committed:
style: update references for Figma

ju-Skinner committed:
style: refactor border-radius to dimension

also change border-radius semantic to only use full

dev-kjbot committed:
style: token studio changes

dev-kjbot committed:
style: add new typography tokens for body and body-sm

dev-kjbot committed:
style: update line-height to unit-less

For typography styles a calculation was done to create in the form of percentages

ju-Skinner committed:
style(fonts): fix misspelling of sprig filename variable

dev-kjbot committed:
chore: fixes success typo and adds select component token

Commit:
chore: fixes components with outdated tokens

Commit:
chore: small component fixes

Commit:
chore: fixes tests
